app/Http/Controllers/GuestController.php

```diff
@@ -129,22 +129,34 @@ public function postRating(Request $request)
         //     // the alternative.
         // }
     }
+    public function showCheckout(Request $request)
+    {
+        return view('guest.checkout');
+    }
+
+    public function showCheckoutSuccess(Request $request)
+    {
+        return view('guest.checkout.success');
+    }
 
-    public function tearDown($input)
+    public function tearDown(Request $request)
     {
-        $input = $input;
         // get hash
-        // TODO: get guest id from input
-        $hash = Guest::find(1)->generateUserHash();
+        // TODO: get guest id from cookie
+        $roomNumber = $request->input('roomNumber');
+        $hash = Guest::where('roomNumber', $roomNumber)->first()->generateUserHash();
         // compare database entries to hash
-        $dbHash = DB::table('ratings')->select('id')->where('user_hash', $hash)->get();
-        print_r($dbHash);
+        $dbHash = DB::table('ratings')
+                    ->select('id')
+                    ->where('user_hash', $hash)
+                    ->get();
+        //print_r($dbHash);
         // strip hashes from matching entries
         foreach ($dbHash as $f) {
             $id = $f->id;
             DB::table('ratings')
                 ->where('id', $id)
-                ->update(['user_hash' => 'removed after checkout']);
+                ->update(['user_hash' => null]);
         }
         // redirect to success page
         return redirect('guest/checkout/success');
```

Comment:
Redirect should be to `/admin/guest/checkout/success`, it now points to `/guest/checkout/success`.

Reply:
Use the named route `guest.checkout.success`.